src/models/utils.ts

```diff
@@ -234,15 +237,16 @@ export function isSameProfile(currentProfile: interfaces.IConnectionProfile, exp
 
 /**
  * Compares 2 connections to see if they match. Logic for matching:
- * match on all key properties (server, db, auth type, user) being identical.
+ * match on all key properties (connectionString or server, db, auth type, user) being identical.
  * Other properties are ignored for this purpose
  *
  * @param {IConnectionCredentials} conn the connection to check
  * @param {IConnectionCredentials} expectedConn the connection to try to match
  * @returns boolean that is true if the connections match
  */
 export function isSameConnection(conn: interfaces.IConnectionCredentials, expectedConn: interfaces.IConnectionCredentials): boolean {
-    return expectedConn.server === conn.server
+    return (conn.connectionString || expectedConn.connectionString) ? conn.connectionString === expectedConn.connectionString :
```

Comment:
I think this should be ``(conn.connectionString && expectedConn.connectionString)`` in case ``conn`` or ``expectedConn`` is null.